make clean breaks build

I installed and built the repo with

```
git clone....
cd silkrpc
mkdir -p build && cd build
cmake ..
make
```
The build works fine... I then did these commands:

```
make clean
make
```

As soon as the build starts, I get this warning:

```
sh@desktop:~/D/T/build|main✓➤ make
-- GLOB mismatch!
-- GLOB mismatch!
-- GLOB mismatch!
-- GLOB mismatch!
```

and the link fails with this error:

```
[100%] Linking CXX executable kv_seek
Undefined symbols for architecture x86_64:
  "remote::KV::Stub::TxRaw(grpc_impl::ClientContext*)", referenced from:
      _main in kv_seek.cpp.o
  "remote::KV::NewStub(std::__1::shared_ptr<grpc::ChannelInterface> const&, grpc::StubOptions const&)", referenced from:
      _main in kv_seek.cpp.o
  "remote::Pair::Pair(google::protobuf::Arena*)", referenced from:
      _main in kv_seek.cpp.o
  "remote::Pair::~Pair()", referenced from:
      _main in kv_seek.cpp.o
  "remote::Cursor::Cursor(google::protobuf::Arena*)", referenced from:
      _main in kv_seek.cpp.o
  "remote::Cursor::~Cursor()", referenced from:
      _main in kv_seek.cpp.o
  "vtable for remote::KV::Stub", referenced from:
      _main in kv_seek.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [cmd/kv_seek] Error 1
make[1]: *** [cmd/CMakeFiles/kv_seek.dir/all] Error 2
make: *** [all] Error 2
```

Without doing anything else, if I then do:

```
make
```
the build works.

In the PR I'm working on, only the `.proto` files are stored in the repo. The generated code is placed in the `./build` folder which is not included in the repo.

I suggest we remove the generated code from the repo (one of the reasons is exactly this issue.)

This allows us to do two things:
1) Edit only `.proto` files and the build will take care of the rest,
2) add new `.proto` files and have them build without modifying the `CMakeLists.txt` files.
3) `make clean`

--- a/README.md
+++ b/README.md
@@ -32,20 +32,51 @@ Building SilkRPC daemon requires
 * [Cpplint](https://github.com/cpplint/cpplint) (`pip3 install cpplint`)
 * [Pylint](https://github.com/PyCQA/pylint) (`pip3 install pylint`)
 
-Once the prerequisites are installed and assuming your [GCC](https://www.gnu.org/software/gcc/) compiler available as `gcc` and `g++` at the command line prompt is at least 10.2.0, bootstrap cmake by running
+Please make your [GCC](https://www.gnu.org/software/gcc/) or [Clang](https://clang.llvm.org/) compiler available as `gcc` and `g++` or `clang` and `clang++` at the command line prompt.
+
+Once the prerequisites are installed, there are convenience [bash](https://www.gnu.org/software/bash/) scripts for complete rebuild and incremental build both in debug and release configurations:
+
+- [GCC](https://www.gnu.org/software/gcc/) compiler
 ```
-mkdir build
-cd build
+# Complete rebuild
+./rebuild_gcc_debug.sh
+./rebuild_gcc_release.sh
+```
+
+```
+# Incremental build
+./build_gcc_debug.sh
+./build_gcc_release.sh
+```
+
+- [Clang](https://clang.llvm.org/) compiler
+```
+# Complete rebuild
+./rebuild_clang_debug.sh
+./rebuild_clang_release.sh
+```
+
+```
+# Incremental build
+./build_clang_debug.sh
+./build_clang_release.sh
+```
+The resulting build folders are `build_[gcc, clang]_[debug, release]` according to your choice.
+
+You can also perform the build step-by-step manually, just bootstrap cmake by running
+```
+mkdir build_gcc_release
+cd build_gcc_release
 cmake ..
 ```
-(BTW, you have to run `cmake ..` just the first time).
+(you have to run `cmake ..` just the first time).
 
 Generate the [gRPC](https://grpc.io/) Key-Value (KV) interface protocol bindings
 ```
 cmake --build . --target generate_kv_grpc
 ```
 
-Then run the build itself
+then run the build itself
 ```
 cmake --build .
 ```
@@ -60,17 +91,10 @@ and check the code style running
 ./run_linter.sh
 ```
 
-There are also convenience [bash](https://www.gnu.org/software/bash/) scripts for a complete rebuild both in debug and release configurations using [GCC](https://www.gnu.org/software/gcc/) compiler (these work even if you have multiple versions installed):
+You can clean the build using
 ```
-./build_gcc_debug.sh
-./build_gcc_release.sh
+cmake --build . --target clean
 ```
-and [Clang](https://clang.llvm.org/) compiler:
-```
-./build_clang_debug.sh
-./build_clang_release.sh
-```
-The resulting build folders are `build_[gcc, clang]_[debug, release]` according to your choice.
 
 ## Windows
 * Install [Visual Studio](https://www.visualstudio.com/downloads) 2019. Community edition is fine.

--- a/silkrpc/commands/eth_api.cpp
+++ b/silkrpc/commands/eth_api.cpp
@@ -52,7 +52,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_block_number(const nlohmann::js
     try {
         ethdb::TransactionDatabase tx_database{*tx};
         const auto block_height = co_await core::get_current_block_number(tx_database);
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(block_height));
+        reply = make_json_content(request["id"], to_quantity(block_height));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -72,7 +72,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_chain_id(const nlohmann::json& 
     try {
         ethdb::TransactionDatabase tx_database{*tx};
         const auto chain_id = co_await core::rawdb::read_chain_id(tx_database);
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(chain_id));
+        reply = make_json_content(request["id"], to_quantity(chain_id));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -87,7 +87,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_chain_id(const nlohmann::json& 
 
 // https://eth.wiki/json-rpc/API#eth_protocolversion
 asio::awaitable<void> EthereumRpcApi::handle_eth_protocol_version(const nlohmann::json& request, nlohmann::json& reply) {
-    reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(common::kETH66));
+    reply = make_json_content(request["id"], to_quantity(common::kETH66));
     co_return;
 }
 
@@ -103,8 +103,8 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_syncing(const nlohmann::json& r
             reply = make_json_content(request["id"], false);
         } else {
             reply = make_json_content(request["id"], {
-                {"currentBlock", "0x" + to_hex_no_leading_zeros(current_block_height)},
-                {"highestBlock", "0x" + to_hex_no_leading_zeros(highest_block_height)},
+                {"currentBlock", to_quantity(current_block_height)},
+                {"highestBlock", to_quantity(highest_block_height)},
             });
         }
     } catch (const std::exception& e) {
@@ -126,7 +126,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_gas_price(const nlohmann::json&
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -231,7 +231,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_block_transaction_count_by_
         const auto block_with_hash = co_await core::rawdb::read_block_by_hash(tx_database, block_hash);
         const auto tx_count = block_with_hash.block.transactions.size();
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(tx_count));
+        reply = make_json_content(request["id"], to_quantity(tx_count));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -264,7 +264,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_block_transaction_count_by_
         const auto block_number = co_await core::get_block_number(block_id, tx_database);
         const auto block_with_hash = co_await core::rawdb::read_block_by_number(tx_database, block_number);
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(block_with_hash.block.transactions.size()));
+        reply = make_json_content(request["id"], to_quantity(block_with_hash.block.transactions.size()));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -392,7 +392,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_uncle_count_by_block_hash(c
         const auto block_with_hash = co_await core::rawdb::read_block_by_hash(tx_database, block_hash);
         const auto ommers = block_with_hash.block.ommers;
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(ommers.size()));
+        reply = make_json_content(request["id"], to_quantity(ommers.size()));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -426,7 +426,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_uncle_count_by_block_number
         const auto block_with_hash = co_await core::rawdb::read_block_by_number(tx_database, block_number);
         const auto ommers = block_with_hash.block.ommers;
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(ommers.size()));
+        reply = make_json_content(request["id"], to_quantity(ommers.size()));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -446,7 +446,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_transaction_by_hash(const n
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -549,7 +549,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_transaction_receipt(const n
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -569,7 +569,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_estimate_gas(const nlohmann::js
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -664,7 +664,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_transaction_count(const nlo
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -684,7 +684,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_storage_at(const nlohmann::
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -745,7 +745,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_new_filter(const nlohmann::json
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -765,7 +765,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_new_block_filter(const nlohmann
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -785,7 +785,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_new_pending_transaction_filter(
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -805,7 +805,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_filter_changes(const nlohma
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -825,7 +825,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_uninstall_filter(const nlohmann
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -960,7 +960,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_send_raw_transaction(const nloh
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -980,7 +980,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_send_transaction(const nlohmann
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -1000,7 +1000,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_sign_transaction(const nlohmann
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -1020,7 +1020,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_proof(const nlohmann::json&
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -1040,7 +1040,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_mining(const nlohmann::json& re
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -1060,7 +1060,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_coinbase(const nlohmann::json& 
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -1080,7 +1080,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_hashrate(const nlohmann::json& 
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -1100,7 +1100,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_submit_hashrate(const nlohmann:
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -1120,7 +1120,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_get_work(const nlohmann::json& 
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -1140,7 +1140,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_submit_work(const nlohmann::jso
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -1160,7 +1160,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_subscribe(const nlohmann::json&
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());
@@ -1180,7 +1180,7 @@ asio::awaitable<void> EthereumRpcApi::handle_eth_unsubscribe(const nlohmann::jso
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(0));
+        reply = make_json_content(request["id"], to_quantity(0));
     } catch (const std::exception& e) {
         SILKRPC_ERROR << "exception: " << e.what() << "\n";
         reply = make_json_error(request["id"], 100, e.what());

--- a/silkrpc/commands/net_api.cpp
+++ b/silkrpc/commands/net_api.cpp
@@ -29,7 +29,7 @@ asio::awaitable<void> NetRpcApi::handle_net_listening(const nlohmann::json& requ
 
 // https://eth.wiki/json-rpc/API#net_peercount
 asio::awaitable<void> NetRpcApi::handle_net_peer_count(const nlohmann::json& request, nlohmann::json& reply) {
-    reply = make_json_content(request["id"], "0x" + to_hex_no_leading_zeros(25));
+    reply = make_json_content(request["id"], to_quantity(25));
     // TODO(canepat): needs p2pSentry integration in TG Core (accumulate peer counts from multiple sentries)
     co_return;
 }

--- a/silkrpc/interfaces/CMakeLists.txt
+++ b/silkrpc/interfaces/CMakeLists.txt
@@ -34,6 +34,8 @@ set(KV_GRPC_HEADERS "${CMAKE_CURRENT_SOURCE_DIR}/remote/kv.grpc.pb.h")
 set(TYPES_PROTO_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/types/types.pb.cc")
 set(TYPES_PROTO_HEADERS "${CMAKE_CURRENT_SOURCE_DIR}/types/types.pb.h")
 
+set_directory_properties(PROPERTIES CLEAN_NO_CUSTOM 1)
+
 add_custom_command(
     OUTPUT "${KV_PROTO_SOURCES}" "${KV_PROTO_HEADERS}" "${KV_GRPC_SOURCES}" "${KV_GRPC_HEADERS}"
     COMMAND ${PROTOBUF_PROTOC}

--- a/silkrpc/json/types.cpp
+++ b/silkrpc/json/types.cpp
@@ -63,6 +63,14 @@ std::string to_hex_no_leading_zeros(uint64_t number) {
     return to_hex_no_leading_zeros(number_bytes);
 }
 
+std::string to_quantity(silkworm::ByteView bytes) {
+    return "0x" + to_hex_no_leading_zeros(bytes);
+}
+
+std::string to_quantity(uint64_t number) {
+    return "0x" + to_hex_no_leading_zeros(number);
+}
+
 } // namespace silkrpc
 
 namespace evmc {
@@ -90,22 +98,22 @@ void from_json(const nlohmann::json& json, bytes32& b32) {
 namespace silkworm {
 
 void to_json(nlohmann::json& json, const BlockHeader& header) {
-    const auto block_number = "0x" + silkrpc::to_hex_no_leading_zeros(header.number);
+    const auto block_number = silkrpc::to_quantity(header.number);
     json["number"] = block_number;
     json["parentHash"] = header.parent_hash;
-    json["nonce"] = "0x" + silkrpc::to_hex_no_leading_zeros({header.nonce.data(), header.nonce.size()});
+    json["nonce"] = silkrpc::to_quantity({header.nonce.data(), header.nonce.size()});
     json["sha3Uncles"] = header.ommers_hash;
     json["logsBloom"] = "0x" + silkworm::to_hex(silkworm::full_view(header.logs_bloom));
     json["transactionsRoot"] = header.transactions_root;
     json["stateRoot"] = header.state_root;
     json["receiptsRoot"] = header.receipts_root;
     json["miner"] = header.beneficiary;
-    json["difficulty"] = "0x" + silkrpc::to_hex_no_leading_zeros(silkworm::rlp::big_endian(header.difficulty));
+    json["difficulty"] = silkrpc::to_quantity(silkworm::rlp::big_endian(header.difficulty));
     json["extraData"] = "0x" + silkworm::to_hex(header.extra_data);
     json["mixHash"]= header.mix_hash;
-    json["gasLimit"] = "0x" + silkrpc::to_hex_no_leading_zeros(header.gas_limit);
-    json["gasUsed"] = "0x" + silkrpc::to_hex_no_leading_zeros(header.gas_used);
-    json["timestamp"] = "0x" + silkrpc::to_hex_no_leading_zeros(header.timestamp);
+    json["gasLimit"] = silkrpc::to_quantity(header.gas_limit);
+    json["gasUsed"] = silkrpc::to_quantity(header.gas_used);
+    json["timestamp"] = silkrpc::to_quantity(header.timestamp);
 }
 
 void to_json(nlohmann::json& json, const Transaction& transaction) {
@@ -113,21 +121,21 @@ void to_json(nlohmann::json& json, const Transaction& transaction) {
         (const_cast<Transaction&>(transaction)).recover_sender();
     }
     json["from"] = transaction.from.value();
-    json["gas"] = "0x" + silkrpc::to_hex_no_leading_zeros(transaction.gas_limit);
-    json["gasPrice"] = "0x" + silkrpc::to_hex_no_leading_zeros(silkworm::rlp::big_endian(transaction.gas_price));
+    json["gas"] = silkrpc::to_quantity(transaction.gas_limit);
+    json["gasPrice"] = silkrpc::to_quantity(silkworm::rlp::big_endian(transaction.gas_price));
     auto ethash_hash{hash_of_transaction(transaction)};
     json["hash"] = silkworm::to_bytes32({ethash_hash.bytes, silkworm::kHashLength});
     json["input"] = "0x" + silkworm::to_hex(transaction.data);
-    json["nonce"] = "0x" + silkrpc::to_hex_no_leading_zeros(transaction.nonce);
+    json["nonce"] = silkrpc::to_quantity(transaction.nonce);
     if (transaction.to) {
         json["to"] =  transaction.to.value();
     } else {
         json["to"] =  "0x";
     }
-    json["value"] =  "0x" + silkrpc::to_hex_no_leading_zeros(silkworm::rlp::big_endian(transaction.value));
-    json["v"] =  "0x" + silkrpc::to_hex_no_leading_zeros(silkworm::rlp::big_endian(transaction.v()));
-    json["r"] =  "0x" + silkrpc::to_hex_no_leading_zeros(silkworm::rlp::big_endian(transaction.r));
-    json["s"] =  "0x" + silkrpc::to_hex_no_leading_zeros(silkworm::rlp::big_endian(transaction.s));
+    json["value"] =  silkrpc::to_quantity(silkworm::rlp::big_endian(transaction.value));
+    json["v"] =  silkrpc::to_quantity(silkworm::rlp::big_endian(transaction.v()));
+    json["r"] =  silkrpc::to_quantity(silkworm::rlp::big_endian(transaction.r));
+    json["s"] =  silkrpc::to_quantity(silkworm::rlp::big_endian(transaction.s));
 }
 
 } // namespace silkworm
@@ -135,32 +143,32 @@ void to_json(nlohmann::json& json, const Transaction& transaction) {
 namespace silkrpc {
 
 void to_json(nlohmann::json& json, const Block& b) {
-    const auto block_number = "0x" + silkrpc::to_hex_no_leading_zeros(b.block.header.number);
+    const auto block_number = silkrpc::to_quantity(b.block.header.number);
     json["number"] = block_number;
     json["hash"] = b.hash;
     json["parentHash"] = b.block.header.parent_hash;
-    json["nonce"] = "0x" + silkrpc::to_hex_no_leading_zeros({b.block.header.nonce.data(), b.block.header.nonce.size()});
+    json["nonce"] = silkrpc::to_quantity({b.block.header.nonce.data(), b.block.header.nonce.size()});
     json["sha3Uncles"] = b.block.header.ommers_hash;
     json["logsBloom"] = "0x" + silkworm::to_hex(silkworm::full_view(b.block.header.logs_bloom));
     json["transactionsRoot"] = b.block.header.transactions_root;
     json["stateRoot"] = b.block.header.state_root;
     json["receiptsRoot"] = b.block.header.receipts_root;
     json["miner"] = b.block.header.beneficiary;
-    json["difficulty"] = "0x" + silkrpc::to_hex_no_leading_zeros(silkworm::rlp::big_endian(b.block.header.difficulty));
-    json["totalDifficulty"] = "0x" + silkrpc::to_hex_no_leading_zeros(silkworm::rlp::big_endian(b.total_difficulty));
+    json["difficulty"] = silkrpc::to_quantity(silkworm::rlp::big_endian(b.block.header.difficulty));
+    json["totalDifficulty"] = silkrpc::to_quantity(silkworm::rlp::big_endian(b.total_difficulty));
     json["extraData"] = "0x" + silkworm::to_hex(b.block.header.extra_data);
     json["mixHash"]= b.block.header.mix_hash;
     silkworm::Bytes block_rlp{};
     silkworm::rlp::encode(block_rlp, b.block);
-    json["size"] = "0x" + silkrpc::to_hex_no_leading_zeros(block_rlp.length());
-    json["gasLimit"] = "0x" + silkrpc::to_hex_no_leading_zeros(b.block.header.gas_limit);
-    json["gasUsed"] = "0x" + silkrpc::to_hex_no_leading_zeros(b.block.header.gas_used);
-    json["timestamp"] = "0x" + silkrpc::to_hex_no_leading_zeros(b.block.header.timestamp);
+    json["size"] = silkrpc::to_quantity(block_rlp.length());
+    json["gasLimit"] = silkrpc::to_quantity(b.block.header.gas_limit);
+    json["gasUsed"] = silkrpc::to_quantity(b.block.header.gas_used);
+    json["timestamp"] = silkrpc::to_quantity(b.block.header.timestamp);
     if (b.full_tx) {
         json["transactions"] = b.block.transactions;
         for (auto i{0}; i < json["transactions"].size(); i++) {
             auto& json_txn = json["transactions"][i];
-            json_txn["transactionIndex"] = "0x" + silkrpc::to_hex_no_leading_zeros(i);
+            json_txn["transactionIndex"] = silkrpc::to_quantity(i);
             json_txn["blockHash"] = b.hash;
             json_txn["blockNumber"] = block_number;
         }
@@ -187,10 +195,10 @@ void to_json(nlohmann::json& json, const Block& b) {
 void to_json(nlohmann::json& json, const Transaction& transaction) {
     to_json(json, silkworm::Transaction(transaction));
 
-    const auto block_number = "0x" + silkrpc::to_hex_no_leading_zeros(transaction.block_number);
+    const auto block_number = silkrpc::to_quantity(transaction.block_number);
     json["blockHash"] = transaction.block_hash;
     json["blockNumber"] = block_number;
-    json["transactionIndex"] = "0x" + silkrpc::to_hex_no_leading_zeros(transaction.transaction_index);
+    json["transactionIndex"] = silkrpc::to_quantity(transaction.transaction_index);
 }
 
 void from_json(const nlohmann::json& json, Call& call) {
@@ -232,11 +240,11 @@ void to_json(nlohmann::json& json, const Log& log) {
     json["address"] = log.address;
     json["topics"] = log.topics;
     json["data"] = "0x" + silkworm::to_hex(log.data);
-    json["blockNumber"] = "0x" + silkrpc::to_hex_no_leading_zeros(log.block_number);
+    json["blockNumber"] = silkrpc::to_quantity(log.block_number);
     json["blockHash"] = log.block_hash;
     json["transactionHash"] = log.tx_hash;
-    json["transactionIndex"] = "0x" + silkrpc::to_hex_no_leading_zeros(log.tx_index);
-    json["logIndex"] = "0x" + silkrpc::to_hex_no_leading_zeros(log.index);
+    json["transactionIndex"] = silkrpc::to_quantity(log.tx_index);
+    json["logIndex"] = silkrpc::to_quantity(log.index);
     json["removed"] = log.removed;
 }
 
@@ -277,14 +285,14 @@ void from_json(const nlohmann::json& json, Log& log) {
 
 void to_json(nlohmann::json& json, const Receipt& receipt) {
     json["blockHash"] = receipt.block_hash;
-    json["blockNumber"] = "0x" + silkrpc::to_hex_no_leading_zeros(receipt.block_number);
+    json["blockNumber"] = silkrpc::to_quantity(receipt.block_number);
     json["transactionHash"] = receipt.tx_hash;
-    json["transactionIndex"] = "0x" + silkrpc::to_hex_no_leading_zeros(receipt.tx_index);
+    json["transactionIndex"] = silkrpc::to_quantity(receipt.tx_index);
     json["from"] = receipt.from.value_or(evmc::address{});
     json["to"] = receipt.to.value_or(evmc::address{});
-    json["type"] = "0x" + silkrpc::to_hex_no_leading_zeros(receipt.type ? receipt.type.value() : 0);
-    json["gasUsed"] = "0x" + silkrpc::to_hex_no_leading_zeros(receipt.gas_used);
-    json["cumulativeGasUsed"] = "0x" + silkrpc::to_hex_no_leading_zeros(receipt.cumulative_gas_used);
+    json["type"] = silkrpc::to_quantity(receipt.type ? receipt.type.value() : 0);
+    json["gasUsed"] = silkrpc::to_quantity(receipt.gas_used);
+    json["cumulativeGasUsed"] = silkrpc::to_quantity(receipt.cumulative_gas_used);
     if (receipt.contract_address) {
         json["contractAddress"] = receipt.contract_address;
     } else {
@@ -292,7 +300,7 @@ void to_json(nlohmann::json& json, const Receipt& receipt) {
     }
     json["logs"] = receipt.logs;
     json["logsBloom"] = "0x" + silkworm::to_hex(silkworm::full_view(receipt.bloom));
-    json["status"] = "0x" + silkrpc::to_hex_no_leading_zeros(receipt.success ? 1 : 0);
+    json["status"] = silkrpc::to_quantity(receipt.success ? 1 : 0);
 }
 
 void from_json(const nlohmann::json& json, Receipt& receipt) {

--- a/silkrpc/json/types.hpp
+++ b/silkrpc/json/types.hpp
@@ -80,6 +80,9 @@ void to_json(nlohmann::json& json, const Error& error);
 std::string to_hex_no_leading_zeros(uint64_t number);
 std::string to_hex_no_leading_zeros(silkworm::ByteView bytes);
 
+std::string to_quantity(uint64_t number);
+std::string to_quantity(silkworm::ByteView bytes);
+
 nlohmann::json make_json_content(uint32_t id, const nlohmann::json& result);
 nlohmann::json make_json_error(uint32_t id, int32_t code, const std::string& message);
 

